### ISSUE TITLE
Make work with Python 3

### ISSUE THREAD

**Issue by ArchimedesPi** [ArchimedesPi/esp8266-luaupload#10]
Currently when run with Python 3, this happens:
````
Traceback (most recent call last):
  File "./luaupload.py", line 89, in <module>
    cli()
  File "/usr/lib/python3.4/site-packages/click/core.py", line 664, in __call__
    return self.main(*args, **kwargs)
  File "/usr/lib/python3.4/site-packages/click/core.py", line 644, in main
    rv = self.invoke(ctx)
  File "/usr/lib/python3.4/site-packages/click/core.py", line 991, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/lib/python3.4/site-packages/click/core.py", line 837, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/lib/python3.4/site-packages/click/core.py", line 464, in invoke
    return callback(*args, **kwargs)
  File "./luaupload.py", line 86, in upload
    esp.upload(f.read(), {'filename': filename})
  File "/home/liam/github/archimedespi/esp8266-luaupload/luaupload/esp8266.py", line 41, in upload
    self.send('file.remove("{file}")\r\n'.format(file=espoptions['filename']))
  File "/home/liam/github/archimedespi/esp8266-luaupload/luaupload/esp8266.py", line 34, in send
    self.connection.write(data)
  File "/usr/lib/python3.4/site-packages/serial/serialposix.py", line 491, in write
    d = to_bytes(data)
  File "/usr/lib/python3.4/site-packages/serial/serialutil.py", line 76, in to_bytes
    b.append(item)  # this one handles int and str for our emulation and ints for Python 3.x
TypeError: an integer is required
````

**Comment by ArchimedesPi** [comment]
This problem stems from the fact that PySerial running under Python 3 requires a bytestring be passed to `.write`. Curently, calls like `connection.write('foo')` are being used - they simply have to be converted to `connection.write('foo'.encode())`.

**Comment by ArchimedesPi** [comment]
Closed via #11 and, more specifically, f5e8a77.
